### PR TITLE
create shift_owner method to prevent other users to edit shifts

### PR DESCRIPTION
**Pull request by saradotlog**
## What type of PR is this? (check all applicable)

- [] Feature
- [] Refactor
- [] Bug Fix
- [x] Optimization
- [] Documentation Update
- [] Other (describe: )

## Description of what PR does
At the moment any user logged as an organization can edit any shift available if navigates to it through the url.
This PR creates a shift_owner method to prevent any user who is not the shift owner to access to the edit shift view.

## Related PRs and/or Issues (if any)
- https://github.com/OurTimeForTech/shiftwork2/issues/13 step 7
-

## Added tests?

- [ ] yes
- [ ] no, because they aren't needed _(please include reasons for why tests aren't needed)_
- [ ] no, because I need help
- [x] no, they will be added later (please create an Issue for it)

## Added to documentation?

- [ ] Yes, project README
- [x] No documentation needed

### CODE REVIEW

File: app/controllers/shifts_controller.rb [hunk 1]
```diff
@@ -88,4 +93,10 @@ def current_org_id
   def shift_params
     params.require(:shift).permit(:shift_open, :shift_role, :shift_description, :shift_start, :shift_end, :shift_pay, :organization_id, :worker_id)
   end
+
+  def shift_owner
```
Comment (msarit) [inline app/controllers/shifts_controller.rb:97]: Hey @saradotlog, there is a cleaner, more Rails-esque way to define this method 😄 

```ruby
def shift_owner?
    @shift.organization_id == current_org_id
end
```

Adding the question mark to the end of the method indicates that the method returns a `true/false`.
In Rails, the last evaluation is automatically returned, so no need for the `if` block 😉 

Question: I looked for where `current_org_id` is defined in the codebase but I couldn't find it. Am I missing something?

**Review by msarit — CHANGES_REQUESTED** [review]
Great implementation @saradotlog!

I left a comment about refactoring the `shift_onwer` method, and also a question.

File: app/controllers/shifts_controller.rb (hunk 1 above)
Comment (saradotlog) [reply app/controllers/shifts_controller.rb:97]: Hi @msarit Thanks a lot for the tip! I actually looked up time ago the difference between a method with and without a "?" but I totally forgot about it, this is a good working example.
Then I reckon we should do the same for the is_owner method?

Comment (saradotlog) [reply app/controllers/shifts_controller.rb:97]: About the `current_org_id` is defined in this same file, in line 88:
```
def current_org_id
    puts "\n\n\nCurrent User ID: #{current_user.id}\n\n\n"
    return Organization.where(user_id: current_user.id).pluck(:id).first
end
```
But it was already there, I didn't define it. I guess I should remove the "puts..." statement as it was for debugging?

**Review by msarit — APPROVED** [review]

**Review by ncbui — APPROVED** [review]
Am redirected with error message when I try to edit a shift as an org that doesn't own it. Good work @saradotlog

**Pull request merged**